precise/iaf_psc_exp_ps_lossless.cpp

```diff
@@ -652,29 +652,41 @@ nest::iaf_psc_exp_ps_lossless::is_spike_( const double dt )
   const double exp_tau_m = numerics::expm1( dt / P_.tau_m_ );
   const double exp_tau_m_s =
     numerics::expm1( dt / P_.tau_m_ - dt / P_.tau_ex_ );
+  const double I_e = V_.y0_before_ + P_.I_e_;
 
-  double g =
-    ( ( V_.a1_ * I_0 * exp_tau_m_s + exp_tau_m * ( V_.a3_ - P_.I_e_ * V_.a2_ )
+  /* Expressions for f and b below are rewritten but equivalent
+     to those given in Krishnan et al. 2018.
+     The expression for g given in the paper as eq.(49) is incorrect.
+     It can instead be constructed as a line through the points (see Fig.6):
+     (I_theta-I_e, V_th) and (i2, f(i2)) where i2=(I_theta-I_e)*exp(dt/tau_s).
+
+     Note that there is a typo in Algorithm 1 and 2 of the paper:
+     g and f are interchanged. (compare to Fig.6) 
+  */
+
+  double f =
```

Comment:
@ChristianKeup Make this `const double`, since `f` does not change below. This allows the compiler to generate more optimal code.